packages/plugins/src/components/plugin-sidebar/index.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import { isEmpty } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, Panel, Slot, Fill } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+//import PinnedPlugins from '../../header/pinned-plugins';
+import { withPluginContext } from '../plugin-context';
+import SidebarHeader from './sidebar-header';
+
+function PluginSidebarPinnedItems( { scope, ...props } ) {
+	return <Fill name={ `PluginPinnedAreas/${ scope }` } { ...props } />;
+}
+
+function PluginSidebarPinnedItemsSlot( { scope, className, ...props } ) {
+	return (
+		<Slot name={ `PluginPinnedAreas/${ scope }` } { ...props }>
+			{ ( fills ) =>
+				! isEmpty( fills ) && (
+					<div className={ className }>{ fills }</div>
+				)
+			}
+		</Slot>
+	);
+}
+
+function PluginSidebarSlot( { scope, ...props } ) {
+	return <Slot name={ `PluginSidebar/${ scope }` } { ...props } />;
+}
+
+function PluginSidebarFill( { scope, ...props } ) {
+	return <Fill name={ `PluginSidebar/${ scope }` } { ...props } />;
+}
+
+function PluginSidebar( {
+	sidebarName,
+	children,
+	className,
+	icon,
+	title,
+	scope,
+	...props
+} ) {
+	const { isActive, isPinned } = useSelect(
+		( select ) => {
+			const { getSingleActiveArea, isMultipleActiveAreaActive } = select(
+				'core/plugins'
+			);
+			return {
+				isActive: getSingleActiveArea( scope ) === sidebarName,
+				isPinned: isMultipleActiveAreaActive( scope, sidebarName ),
```

Comment:
It's a bit confusing to see the names of selectors so loosely related to the local constants used. Can you explain how multiple active areas relate to pinned items? I might be missing something because the notion of active areas is brand new.

For reference, this is how it is implemented in `edit-post` package:

https://github.com/WordPress/gutenberg/blob/8bba7e11ec04e655351b833a0d9a2ea8f3c8cb0b/packages/edit-post/src/components/sidebar/plugin-sidebar/index.js#L149-L150

_Aside:_ to be clear, I don't like `getActiveGeneralSidebarName` at all :)